src/libcollections/vec_deque.rs

```diff
@@ -1829,6 +1811,42 @@ fn wrap_index(index: usize, size: usize) -> usize {
     index & (size - 1)
 }
 
+/// Returns the two slices that cover the VecDeque's valid range
+trait RingSlices : Sized {
+    fn slice(self, from: usize, to: usize) -> Self;
+    fn split_at(self, i: usize) -> (Self, Self);
+
+    fn ring_slices(buf: Self, head: usize, tail: usize) -> (Self, Self) {
```

Comment:
Maybe some `#[inline]`-s here and on the `slice`/`split_at` methods below would help ensure the generated code stays the same as before? (don't have any data that it doesn't and since code is generic, it'll always be available to inline)